Homework button

hard-working opened the pull request.

hard-working committed:
redefine colors by style guide

hard-working committed:
add button styles

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.

GSemikozov commented on 03-lection1/01-button/button.css:
```diff
@@ -1 +1,66 @@
+  border: 1px solid var(--border-color);
+}
+
+.button:hover {
```
видимо .button_primary?

GSemikozov commented on 03-lection1/01-button/button.css:
```diff
@@ -1 +1,66 @@
+  min-width: 189px;
+  min-height: 48px;
+  padding: 15px 65px 14px;
+  background-color: var(--primary);
```
фон и текст - это уже не общие для всех типов стили, в данном случае они относятся к .button_primary

hard-working replied:
Я думал сделать .button кнопкой по умолчанию, когда она без модификаторов, то имеет стиль основной, primary кнопки. Поправил по замечаниям, но не понял, почему прошлый вариант - плохо. Теперь, если задать стиль кнопке без модификатора, она будет выделяться из общего дизайна.

GSemikozov replied:
у нас есть конкретные типы кнопок, не нужно создавать кнопки по дефолту которые не типизированы в дизайне. По дефолту можно сделать например другой фон, чтобы кнопка просто выглядела как не стилизованная. Но в целом - primary это гораздо понятнее, что это основная кнопка, и нет необходимости думать о том, будет ли .button без спецификации типа кнопкой primary или еще какой-то

GSemikozov commented on 03-lection1/01-button/button.css:
```diff
@@ -1 +1,66 @@
+
+.button:disabled {
+  opacity: var(--primary-opacity);
+  cursor: default;
```
чтобы пользователь не мог нажимать лучше добавить pointer-events: none

hard-working committed:
change button:disabled

add pointer-events: none
remove needless props

hard-working committed:
add button_primary modifier

move bg-color & color from .button to .button_primary
add button_primary class to default button
